miner/worker.go

```diff
@@ -737,7 +737,14 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 	}
 
 	var coalescedLogs []*types.Log
-
+	var stopTimer *time.Timer
+	delay := w.engine.Delay(w.chain, w.current.header)
+	if delay != nil {
+		stopTimer = time.NewTimer(*delay - w.config.DelayLeftOver)
```

Comment:
Is it possible that `*delay - w.config.DelayLeftOver` is negative?

Reply:
Yes, if it is negative, the channel will receive a signal immediately. Just tested it.